### ISSUE TITLE
Add the possibility to search for files in the packages, like a command name 

### ISSUE THREAD

**Issue by mickours** [NixOS/nixos-homepage#216]
It would interesting to have a page that permits the users to search for a file in the packages. A good example is a program that you need but you only now the executable name that differ from the package name. For example, I was looking for the `readelf` command and I was not able to find it from the Packages search page or the `nix search` tool.  
 
Debian has a good tool for that:
https://www.debian.org/distrib/packages#search_contents
The readelf example:
https://packages.debian.org/search?searchon=contents&keywords=%2Fbin%2Freadelf&mode=path&suite=stable&arch=any

It would be very nice to have a similar page in the Nixos site.

**Comment by samueldr** [comment]
Agreed, though this would need server-side work to happen (or some service hosted elsewhere).

All current search pages on the website are completely handled client-side by having an index loaded from the server.

Using [`nix-index`](https://github.com/bennofs/nix-index) you can locally search for files for any package (as long as it's built on hydra). It may be possible to build a service that, as a backend, would use `nix-index` to return results.

**Comment by mickours** [comment]
Also, [command-not-found](https://github.com/NixOS/nixpkgs/tree/master/nixos/modules/programs/command-not-found) seems to do the job locally:
```sh
$ readelf
The program ‘readelf’ is currently not installed. You can install it by typing:
  nix-env -iA nixos.binutils-unwrapped
```

I'm using Zsh and the support was broken at the time I've created this issue.
Still, the idea remain interesting for systems that do not have ``command-not-found`` activated. 
 
Maybe the ``command-not-found`` script can be adapted to be use as backend for the website as an alternative to ``nix-index``.
